pkg/platform/runtime/buildscript/json.go

```diff
@@ -0,0 +1,91 @@
+package buildscript
+
+import (
+	"bytes"
+	"io"
+)
+
+func (s *Script) ToJson() []byte {
+	json := &bytes.Buffer{}
+	s.WriteJson(json)
+	return json.Bytes()
+}
+
+func writeString(w io.Writer, s string) {
+	w.Write([]byte(`"`))
+	w.Write([]byte(s))
+	w.Write([]byte(`"`))
+}
+
+func (s *Script) WriteJson(w io.Writer) {
+	w.Write([]byte(`{"let":{`))
+	for i, assignment := range s.Let.Assignments {
+		assignment.WriteJson(w)
+		if i+1 < (len(s.Let.Assignments)) {
+			w.Write([]byte(","))
+		}
+	}
+	w.Write([]byte(`,"in":`))
+	switch {
+	case s.In.FuncCall != nil:
+		s.In.FuncCall.WriteJson(w)
+	case s.In.Name != nil:
+		writeString(w, "$"+*s.In.Name)
+	}
+	w.Write([]byte(`}}`))
+}
+
+func (a *Assignment) WriteJson(w io.Writer) {
+	writeString(w, a.Key)
+	w.Write([]byte(":"))
+	a.Value.WriteJson(w)
+}
```

Comment:
Same as with the Write method, implementing different interfaces should make for more flexible usage. In this case, implementing json.Marshaler (https://pkg.go.dev/encoding/json#Marshaler) should result in much simpler handling when getting JSON output from a type. This might mean that the marshaling requires a call to the json package, but that addition should be minor compared to the reductions.

Reply:
Thanks for the pointer. I'll use the marshaler interface.